tests/cpp-tests/Source/MainLoopTest/MainLoopTest.cpp

```diff
@@ -0,0 +1,133 @@
+    throttleLabel->setPosition(window.width / 2, window.height - 80);
+    addChild(throttleLabel);
+
+    auto fpsLabel = Label::createWithTTF("Fps: 60", "fonts/arial.ttf", 18);
+    fpsLabel->setPosition(window.width / 2, window.height - 100);
+    addChild(fpsLabel);
+
+    auto sprite = Sprite::create("Images/blocks.png");
+    sprite->setPosition(window.width / 2 + 150, window.height / 2 + 100);
+    addChild(sprite);
+    sprite->runAction(RepeatForever::create(RotateBy::create(1.0f, 90.0f)));
+
+    auto addFpsButton = [=](int col, int row, const std::string& text, const std::function<void()>& callback)
+    {
+        auto label = Label::createWithTTF(text, "fonts/arial.ttf", 20);
+        auto item  = MenuItemLabel::create(label, [=](Object*)
+        {
+            callback();
+            fpsLabel->setString(StringUtils::format("Fps: %d", sFps));
+            if (sFps == 0)
+                Director::getInstance()->setAnimationInterval(0.0f);
+            else
+                Director::getInstance()->setAnimationInterval(1.0f / float(sFps));
+        });
+        item->setPosition(left + col * colWidth, top - row * rowHeight);
+        menuRequest->addChild(item);
+    };
+
+    auto addUpdateButton = [=](int col, int row, const std::string& text, const std::function<void()>& callback)
+    {
+        auto label = Label::createWithTTF(text, "fonts/arial.ttf", 20);
+        auto item  = MenuItemLabel::create(label, [=](Object*)
+        {
+            callback();
+            throttleLabel->setString(StringUtils::format("Throttle: %dms", sUpdateThrottleMs));
+        });
+        item->setPosition(left + col * colWidth, top - row * rowHeight);
+        menuRequest->addChild(item);
+    };
+
+    addFpsButton(0, 0, "Reset", [this] { sFps = 0; });
+    addFpsButton(0, 1, "-1", [this] { sFps -= 1; });
+    addFpsButton(1, 1, "+1", [this] { sFps += 1; });
+    addFpsButton(0, 2, "-10", [this] { sFps -= 10; });
+    addFpsButton(1, 2, "+10", [this] { sFps += 10; });
+    addFpsButton(0, 3, "15", [this] { sFps = 15; });
+    addFpsButton(1, 3, "30", [this] { sFps = 30; });
+    addFpsButton(0, 4, "60", [this] { sFps = 60; });
+    addFpsButton(1, 4, "120", [this] { sFps = 120; });
+
+    addUpdateButton(2, 0, "Reset", [this] { sUpdateThrottleMs = 0; });
+    addUpdateButton(2, 1, "-1ms", [this] { sUpdateThrottleMs -= 1; });
+    addUpdateButton(3, 1, "+1ms", [this] { sUpdateThrottleMs += 1; });
+    addUpdateButton(2, 2, "-10ms", [this] { sUpdateThrottleMs -= 10; });
+    addUpdateButton(3, 2, "+10ms", [this] { sUpdateThrottleMs += 10; });
+    addUpdateButton(2, 3, "16ms", [this] { sUpdateThrottleMs = 16; });
+    addUpdateButton(3, 3, "33ms", [this] { sUpdateThrottleMs = 33; });
+    addUpdateButton(2, 4, "50ms", [this] { sUpdateThrottleMs = 50; });
+    addUpdateButton(3, 4, "100ms", [this] { sUpdateThrottleMs = 100; });
+
+    scheduleUpdate();
+}
+
+void MainLoopTest::update(float dt)
```

Comment:
Sorry I do not clearly understand what about this fix? Does it help to get requested fps for high frequency devices? 

Why such implementation of update? As I see:
1. Remember now time with some Delta
2. Wait while this Delta expiration nothing doing

You can use Node::schedule with selector for this purpose?

Reply:
This should fix the issue of requested animation interval not working on devices with high refresh rate displays. And it also fixes frames not starting on vsync.

The `MainLoopTest::update()` thing is the simplest thing to add simulation of slow frame updates.